### ISSUE TITLE
Configuration Errror - "Cannot connect to Strava API"

### ISSUE THREAD

**Issue by AliNjie** [grafana/strava-datasource#5]
I am having issues getting the Strava datasource to work properly. When saving and testing the datasource, this happens:
![image](https://user-images.githubusercontent.com/38862494/76438508-5ee01780-63bb-11ea-9705-8666e9356c60.png)

Related messages from the Grafana server log
```
t=2020-03-11T15:58:46+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="Could not find executor for data source type: grafana-strava-datasource"
t=2020-03-11T15:58:46+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/tsdb/query status=500 remote_addr=10.0.0.17 time_ms=5 size=34 referer="http://10.0.0.100:3000/datasources/edit/3/?state=&code=<INSERT_CODE_HERE>&scope=read,activity:read_all,profile:read_all,read_all"
```

Grafana Version: `Version 6.6.2 (commit: 3fa63cfc34, branch: HEAD)`

Any help would be greatly appriciated!

**Comment by alexanderzobnin** [comment]
Hi! Did you configure datasource according to [docs](https://github.com/grafana/strava-datasource/blob/master/docs/configuration.md)?

**Comment by alexanderzobnin** [comment]
Also, `Could not find executor for data source type: grafana-strava-datasource` means backend is not started. Please, make sure you have plugin properly installed.

**Comment by AliNjie** [comment]
HI @alexanderzobnin! Thank you for a quick reply.

I've tried to install the plugin as described in the documentation:
I'm starting with an empty plugins list
```
$ sudo grafana-cli plugins ls

Restart grafana after installing plugins . <service grafana-server restart>
```
I then install the plugin
```
$ sudo grafana-cli plugins install grafana-strava-datasource
installing grafana-strava-datasource @ 1.1.1
from: https://grafana.com/api/plugins/grafana-strava-datasource/versions/1.1.1/download
into: /var/lib/grafana/plugins

✔ Installed grafana-strava-datasource successfully

Restart grafana after installing plugins . <service grafana-server restart>
```

I then make sure that the directory and environment variable is created:
```
$ sudo mkdir /var/lib/grafana/strava
$ export GF_STRAVA_DS_DATA_PATH=/var/lib/grafana/strava
```

After this, I restart the server with `sudo systemctl restart grafana-server` and follow the process in the docs you linked, but run in to the same error.

**Comment by AliNjie** [comment]
The Grafana server is running on a Raspberry Pi, if that is of any relevance
```
$ uname -a
Linux raspberrypi 4.19.75-v7+ #1270 SMP Tue Sep 24 18:45:11 BST 2019 armv7l GNU/Linux
```

**Comment by alexanderzobnin** [comment]
Plugin doesn't have compiled binaries for `arm` arch. If you want to run it with Raspberry Pi, you should build it for this arch first.

**Comment by AliNjie** [comment]
Ah, okay. Makes sense.

I’m not familiar with how to build it for a specific architecture. Do you have any suggestions as to how I should approach it?

**Comment by Chrisblueeyes83** [comment]
please help me with the variable $ export GF_STRAVA_DS_DATA_PATH=/var/lib/grafana/strava
i odnt know how to create this...

**Comment by BeWog** [comment]
In /etc/default/grafana-server file, add this line:
`GF_STRAVA_DS_DATA_PATH=/var/lib/grafana/strava`

Then I "Save & Test", connection is working but dashboard doesnt not work. 
The same error come back:
> Cannot connect to Strava API

**Comment by Chrisblueeyes83** [comment]
that didnt work for me :(

**Comment by Chrisblueeyes83** [comment]
please, will anybody help me with this problem? this drives me crazy!

**Comment by BeWog** [comment]
Do you have last version of Grafana and plugins ?
Please explain ;-)

**Comment by Chrisblueeyes83** [comment]
yes, its the latest version.
the process to register works fine, but ~5min later i have to re activate.

**Comment by Tigermatze** [comment]
> Ah, okay. Makes sense.
> 
> I’m not familiar with how to build it for a specific architecture. Do you have any suggestions as to how I should approach it?

Can anybody explain how we can build it for an ARM achitecture?

**Comment by nzrunner** [comment]
I have the same issue as @Chrisblueeyes83. The Data Source keeps disconnecting from Strava.

Any solution for that?

**Comment by Chrisblueeyes83** [comment]
after an general update the datasource keeps connected. dont know why....

**Comment by AliNjie** [comment]
closed by #6

**Comment by nzrunner** [comment]
> after an general update the datasource keeps connected. dont know why....

What do you mean by "general update"? Mine is still disconnecting.
